multicluster/controllers/multicluster/core/resourceimport_controller.go

```diff
@@ -0,0 +1,411 @@
+		err := r.localClusterClient.Create(ctx, mcsEpObj, &client.CreateOptions{})
+		if err != nil {
+			klog.ErrorS(err, "fail to create MCS Endpoints", "endpoints", klog.KObj(mcsEpObj), err)
+			return ctrl.Result{}, err
+		}
+		r.installedResImports.Add(*resImp)
+		return ctrl.Result{}, nil
+	}
+	if _, ok := ep.Annotations[common.AntreaMCServiceAnnotation]; !ok {
+		klog.InfoS("Endpoints has no desired annotation "+common.AntreaMCServiceAnnotation+", skip update", "endpoints", epNamespaced.String())
+		return ctrl.Result{}, nil
+	}
+	// TODO: check label difference ?
+	if !apiequality.Semantic.DeepEqual(newSubsets, ep.Subsets) {
+		ep.Subsets = newSubsets
+		err = r.localClusterClient.Update(ctx, ep, &client.UpdateOptions{})
+		if err != nil {
+			klog.ErrorS(err, "fail to update MCS Endpoints", "endpoints", epNamespaced.String())
+			return ctrl.Result{}, err
+		}
+		r.installedResImports.Update(*resImp)
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *ResourceImportReconciler) handleResImpDeleteForEndpoints(ctx context.Context, resImp *multiclusterv1alpha1.ResourceImport) (ctrl.Result, error) {
+	epName := common.AntreaMCSPrefix + resImp.Spec.Name
+	epNamespaced := types.NamespacedName{Namespace: resImp.Spec.Namespace, Name: epName}
+	klog.InfoS("Deleting Endpoints corresponding to ResourceImport", "endpoints", epNamespaced.String(),
+		"resourceimport", klog.KObj(resImp))
+
+	ep := &corev1.Endpoints{}
+	err := r.localClusterClient.Get(ctx, epNamespaced, ep)
+	if err != nil {
+		klog.InfoS("unable to fetch imported Endpoints", "endpoints", epNamespaced.String(), "err", err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	err = r.localClusterClient.Delete(ctx, ep, &client.DeleteOptions{})
+	if err != nil {
+		klog.InfoS("fail to delete imported Endpoints", "endpoints", epNamespaced.String(), "err", err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	return ctrl.Result{}, nil
+}
+
+func getMCService(resImp *multiclusterv1alpha1.ResourceImport) *corev1.Service {
+	mcsPorts := []corev1.ServicePort{}
+	for _, p := range resImp.Spec.ServiceImport.Spec.Ports {
+		mcsPorts = append(mcsPorts, corev1.ServicePort{
+			Name:     p.Name,
+			Port:     p.Port,
+			Protocol: p.Protocol,
+		})
+	}
+	mcs := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      common.AntreaMCSPrefix + resImp.Spec.Name,
+			Namespace: resImp.Spec.Namespace,
+			Annotations: map[string]string{
+				common.AntreaMCServiceAnnotation: "true",
+				common.SourceImportAnnotation:    resImp.Name,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Type:  corev1.ServiceTypeClusterIP,
+			Ports: mcsPorts,
+		},
+	}
+	return mcs
+}
+
+func getMCServiceImport(resImp *multiclusterv1alpha1.ResourceImport) *k8smcsv1alpha1.ServiceImport {
+	svcImp := &k8smcsv1alpha1.ServiceImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resImp.Spec.Name,
+			Namespace: resImp.Spec.Namespace,
+			Annotations: map[string]string{
+				common.SourceImportAnnotation: resImp.Name,
```

Comment:
@jianjuns  the resImp.Name format is "$namespace_$name_$kind", eg: `default-nginx-endpoints` and `default-nginx-service`, its original purpose of this annotation is for MC Service clean up, it might not be used anymore, I will double check this.